source/core/parameters.cc

```diff
@@ -844,27 +844,61 @@ namespace Parameters
 
       std::string op;
       op = prm.get("beam orientation");
-      if (op == "x")
+      if (op == "x+")
         {
-          beam_orientation                   = BeamOrientation::x;
+          beam_direction                     = 1;
+          beam_orientation                   = BeamOrientation::positive_x;
           beam_orientation_coordinate        = 0;
           perpendicular_plane_coordinate_one = 1;
           if constexpr (dim == 3)
             perpendicular_plane_coordinate_two = 2;
         }
-      else if (op == "y")
+      else if (op == "x-")
         {
-          beam_orientation                   = BeamOrientation::y;
+          beam_direction                     = 0;
+          beam_orientation                   = BeamOrientation::negative_x;
+          beam_orientation_coordinate        = 0;
+          perpendicular_plane_coordinate_one = 1;
+          if constexpr (dim == 3)
+            perpendicular_plane_coordinate_two = 2;
+        }
+      else if (op == "y+")
+        {
+          beam_direction                     = 1;
+          beam_orientation                   = BeamOrientation::positive_y;
+          perpendicular_plane_coordinate_one = 0;
+          beam_orientation_coordinate        = 1;
+          if constexpr (dim == 3)
+            perpendicular_plane_coordinate_two = 2;
+        }
+      else if (op == "y-")
+        {
+          beam_direction                     = 0;
+          beam_orientation                   = BeamOrientation::negative_y;
           perpendicular_plane_coordinate_one = 0;
           beam_orientation_coordinate        = 1;
           if constexpr (dim == 3)
             perpendicular_plane_coordinate_two = 2;
         }
-      else if (op == "z")
+      else if (op == "z+")
+        {
+          if constexpr (dim == 3)
+            {
+              beam_direction                     = 1;
+              beam_orientation                   = BeamOrientation::positive_z;
+              perpendicular_plane_coordinate_one = 0;
+              perpendicular_plane_coordinate_two = 1;
+              beam_orientation_coordinate        = 2;
+            }
+          else if constexpr (dim == 2)
+            Assert(dim == 2, TwoDimensionalLaserError(dim));
```

Comment:
Maybe it would be interesting if it was working in 2D for geometries in the xz or yz planes.

Reply:
Good question. Since the current laser model is 3d (it has penetration depth as a parameter), in 2d simulations we need to have a laser beam compatible with the grid, and we can't have beam emissions perpendicular to the 2d plane.